qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1426,3 +1422,53 @@ def _update_accession_numbers(self, column, values):
             else:
                 warnings.warn("No new accession numbers to update",
                               qdb.exceptions.QiitaDBWarning)
+
+    def validate(self, restriction_dict):
+        """ Validate the values in the restricted fields in info files
+
+        Parameters
+        ----------
+        restriction_dict : dict of {str: Restriction}
+            A dictionary with the restrictions that apply to the metadata
+
+        Raises
+        ------
+        QiitaDBWarning
+            If the values aren't castable
+        """
+        warning_msg = []
+        columns = self.categories()
+        for label, restriction in viewitems(restriction_dict):
+            missing = set(restriction.columns).difference(columns)
+            if missing:
+                warning_msg.append(
+                    "%s: %s" % (restriction.error_msg,
+                                ', '.join(sorted(missing))))
+            else:
+                valid_null = qdb.metadata_template.constants.EBI_NULL_VALUES
+                for column, datatype in viewitems(restriction.columns):
+                    for sample, val in viewitems(self.get_category(column)):
+                        # ignore if valid null value
+                        if val in valid_null:
+                            continue
+                        # test values
+                        if datatype == 'datetime':
+                            val = str(val)
+                            try:
+                                datetime.strptime(val, '%m/%d/%y %H:%M:%S')
```

Comment:
I would like the input from @ackermag, if I recall correctly the column `collection_timestamp` was going to change and it can store either a full timestamp or just the date.